src/post/post.service.ts

```diff
@@ -0,0 +1,206 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Post } from 'src/entities/post.entity';
+import { User } from 'src/entities/user.entity';
+import { ArrayContains, DeepPartial, LessThan, MoreThan, Repository } from 'typeorm';
+import { TagService } from './tag.service';
+
+@Injectable()
+export class PostService {
+  constructor(
+    @InjectRepository(Post)
+    private readonly postRepository: Repository<Post>,
+    private readonly tagService: TagService
+  ){}
+  /**
+   * @param definition
+   * post object
+   * @param user
+   * user object from request
+   * @returns Promise
+   */
+  async createPost(definition: DeepPartial<Post>, user: DeepPartial<User>) : Promise<Post> {
+
+    const { title, description, picture, coverPicture, tags } = definition;
+
+    const newPost = this.postRepository.create({
+      title, 
+      description, 
+      picture, 
+      coverPicture, 
+      tags,
+      author: user,
+      metadata:  {
+        viewsCount: 0,
+        likesCount: 0,
+        savesCount: 0
+      }
+    });
+
+    if(tags){
```

Comment:
firstly you need to create post and then tags
and try to do it without this method onCreatePost


one to many should create it automatically